samza-core/src/main/java/org/apache/samza/metrics/MetricGroup.java

```diff
@@ -44,6 +44,10 @@ public Counter newCounter(String name) {
     return registry.newCounter(groupName, (prefix + name).toLowerCase());
   }
 
+  public <T> ListGauge newListGauge(String name) {
+    return registry.newListGauge(groupName, new ListGauge(name));
```

Comment:
should you do a `ListGauge<T>` for explicit type inference here since the method is parameterized by `T` already?